syntax = "proto3";

import "alice/protos/api/matrix/scheduled_action.proto";

option go_package = "a.yandex-team.ru/alice/protos/api/matrix";

package NMatrix.NScheduler.NApi;

// [BEGIN TAddScheduledActionRequest]
message TAddScheduledActionRequest {
    enum EOverrideMode {
        // If scheduled action exists the request will end with an error
        NONE = 0;

        // Override meta and spec but do not touch status (the time of the next perform of the action and ConsecutiveFailuresCounter will not change for example)
        // Useful for periodic actions
        META_AND_SPEC_ONLY = 1;

        // Override all (in fact remove + create)
        ALL = 2;
    };

    // WARNING: Guid is readonly field and must be empty
    NMatrix.NApi.TScheduledActionMeta Meta = 1;
    NMatrix.NApi.TScheduledActionSpec Spec = 2;

    EOverrideMode OverrideMode = 3;
}
// [END TAddScheduledActionRequest]

// [BEGIN TAddScheduledActionResponse]
message TAddScheduledActionResponse {
}
// [END TAddScheduledActionResponse]

// [BEGIN TRemoveScheduledActionRequest]
message TRemoveScheduledActionRequest {
    string ActionId = 1;
}
// [END TRemoveScheduledActionRequest]

// [BEGIN TRemoveScheduledActionResponse]
message TRemoveScheduledActionResponse {
}
// [END TRemoveScheduledActionResponse]
